src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -244,13 +244,24 @@ def __init__(
                 self.addNode(node)
 
         for gen_kw in gen_kw_list:
-            gen_kw_node = self.gen_kw_node(gen_kw, tag_format)
             self._create_node_metainfo(gen_kw, 4, EnkfVarType.PARAMETER)
-            self._config_node_meta[gen_kw[0]].output_file = gen_kw[2]
-            init_file = self._config_node_meta[gen_kw[0]].init_file
+            gen_kw_key = gen_kw[0]
+            self._config_node_meta[gen_kw_key].output_file = gen_kw[2]
+            init_file = self._config_node_meta[gen_kw_key].init_file
             if init_file:
-                self._config_node_meta[gen_kw[0]].init_file = os.path.abspath(init_file)
-            self.addNode(gen_kw_node)
+                self._config_node_meta[gen_kw_key].init_file = os.path.abspath(
+                    init_file
+                )
+
+            kw_node = GenKwConfig(
+                key=gen_kw_key,
+                template_file=_get_abs_path(gen_kw[1]),
+                parameter_file=_get_abs_path(gen_kw[3]),
+                tag_fmt=tag_format,
+            )
+
+            self._check_config_node(kw_node)
+            self.add_gen_kw_node(kw_node)
```

Comment:
It seems that `isinstance(key, list)` is always true in tests. Is it the case that summary_list will only contain lists? In that case it could possibly simplify the logic here.

Reply:
We are postponing this for a later time.